src/main/java/gregtech/api/recipes/GTRecipeInputCache.java

```diff
@@ -0,0 +1,63 @@
+package gregtech.api.recipes;
+
+import gregtech.api.GTValues;
+import gregtech.api.recipes.ingredients.GTRecipeInput;
+import gregtech.api.util.GTLog;
+import gregtech.common.ConfigHolder;
+import it.unimi.dsi.fastutil.objects.ObjectOpenHashSet;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class GTRecipeInputCache {
+
+    private static final int EXPECTED_CACHE_SIZE = 16384;
```

Comment:
Where does this expected cache size value come from?

Reply:
ObjectOpenHashSet's initial array size is always a multiple of two. GregTech caches a little over 10K elements, so it's just a next multiple of two of that.

Previously the initial array size specified was `15072`, which would be bumped up to `32768` since the value is divided by load factor of `.75`. I don't know much about the origin or reasoning behind previous value.